xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAEnumeratorHD.cpp

```diff
@@ -733,3 +546,227 @@ ComPtr<ID3D11VideoProcessorOutputView> CEnumeratorHD::CreateVideoProcessorOutput
 
   return outputView;
 }
+
+ProcessorConversions CEnumeratorHD::QueryHDRConversions(const bool& isSourceFullRange)
+{
+  std::unique_lock<CCriticalSection> lock(m_section);
+
+  // Not initialized yet
+  if (!m_pEnumerator)
+    return {};
+
+  // HDR support started with Windows 10 implementing the Enumerator1 interface
+  if (!m_pEnumerator1)
+  {
+    CLog::LogF(LOGWARNING,
+               "ID3D11VideoProcessorEnumerator1 required to evaluate support is not available.");
+    return {};
+  }
+
+  // There are no full range equivalents of DXGI_COLOR_SPACE_YCBCR_STUDIO_G2084_TOPLEFT_P2020 / LEFT
+  // => exit early as unsupported.
+  if (isSourceFullRange)
+    return {};
+
+  // Top left chroma siting is always preferred for HDR because is used in UHD Blu-Rays
+  // ffmpeg flags of the source can be missing of bad anyway
+  DXGI_COLOR_SPACE_TYPE inputCS = DXGI_COLOR_SPACE_YCBCR_STUDIO_G2084_TOPLEFT_P2020;
+
+  const DXGI_COLOR_SPACE_TYPE destColor = DX::Windowing()->UseLimitedColor()
+                                              ? DXGI_COLOR_SPACE_RGB_STUDIO_G2084_NONE_P2020
+                                              : DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020;
+  const DXGI_COLOR_SPACE_TYPE outputCS = destColor;
+
+  const ProcessorConversions& convTopLeft =
```

Comment:
Would appreciate input of C++ guru as well, is this a way to make a "move" happen, since the function returning the object no longer needs it? instead of using the copy constructor on the returned object.

Reply:
https://en.wikipedia.org/wiki/Copy_elision#Return_value_optimization
https://stackoverflow.com/questions/12953127/what-are-copy-elision-and-return-value-optimization

Reply:
Good links, thanks. I took out this & and its friends in similar calls.
The elision is not mandatory though because this case is NRVO, not RVO.
With our Visual Studio and our project options it means that the optimization is applied for Release build and not Debug. Not a big deal, and it's not the only place this happens, but something to know when troubleshooting weird constructor/destructor stuff.